src/I18nService.js

```diff
@@ -36,39 +36,31 @@ export class I18nService extends EventEmitter {
     this.ensureSupportedLanguage(language)
 
     const dictionary = this.getDictionary(language)
-    if (mark !== undefined && !isNaN(args?.[mark])) {
-      const marked = args?.[mark]
-      let found = null
-      for (let i = 0; i < markedRules.length; i++) {
-        if (
-          dictionary &&
-          dictionary[`${label}.${markedRules[i].suffix}`] &&
-          markedRules[i].condition(marked)
-        ) {
-          found = markedRules[i].suffix
-          break
-        }
-      }
 
-      if (found) {
-        return this.replacePlaceholders(dictionary[`${label}.${found}`], args)
+    const marked = args?.[mark]
+    if (mark && !isNaN(marked)) {
+      const rule = markedRules.find(
+        markedRule =>
+          dictionary?.[`${label}.${markedRule.suffix}`] &&
+          markedRule.condition(marked)
+      )
+
+      if (rule) {
```

Comment:
what is a `rule`, `mark`, `markedRule` ? let's use some well known terms instead like
`key`, `value`, `translation` or something

Reply:
This is out of scope for this PR ... if we want to refactor these terms (and I'm agreeing we should) we need to make a separate task for this

Reply:
this is the perfect time to rename the code you touched .. not code from other files .. not code from other functions .. this code.. this exact 20 lines or so

Reply:
If I have to rename `mark` for example, I have to also update it in the portals (since it is public property)

Reply:
that is a good reason for that one .. but let's do the local ones only

Reply:
`const marked = args?.[pluralBy]`

Reply:
1a52a1211b3bdf7de19cf82eab109b5fbbdf059f